tests/mlio-test/test_text_line_reader.cxx

```diff
@@ -0,0 +1,98 @@
+#include <gtest/gtest.h>
+#include <mlio.h>
+
+namespace mlio {
+
+class test_text_line_reader : public ::testing::Test {
+protected:
+    test_text_line_reader() = default;
+
+    ~test_text_line_reader() override;
+
+protected:
+    std::string const expected_line_1_ = "this is line 1";
+    std::string const expected_line_2_ = "this is line 2";
+    std::string const expected_line_3_ = "this is line 3";
+    std::string const file_path_ = "../resources/test.txt";
+};
+
+test_text_line_reader::~test_text_line_reader() = default;
+
+TEST_F(test_text_line_reader, test_text_line_reader_happy_path)
+{
+    mlio::data_reader_params prm{};
+    prm.dataset.emplace_back(mlio::make_intrusive<mlio::file>(file_path_));
+    prm.batch_size = 3;
+
+    auto reader = mlio::make_intrusive<mlio::text_line_reader>(prm);
+    for (auto i = 0; i < 2; i++) {
+        mlio::intrusive_ptr<mlio::example> exm;
+        while ((exm = reader->read_example()) != nullptr) {
+            auto lbl =
+                dynamic_cast<dense_tensor *>(exm->find_feature("value").get());
```

Comment:
You can/should use a `static_cast` here. You know for sure that "value" is a dense tensor. There is no need to have a runtime check for the type.